api/src/models/Category.ts

```diff
@@ -45,7 +45,11 @@ export class Category {
     }
 
     basename(path: string): string {
-        return path.replace(/\/$/, "").split("/").reverse()[0];
+        return path
```

Comment:
Is there any reason we need all these local basename functions at all? It seems that Node has a native function to do the job: https://nodejs.org/api/path.html#path_path_basename_path_ext